chess-AI/experimental-work/train.py

```diff
@@ -0,0 +1,81 @@
+import random
+import chess
+from mcts import Mcts
+from nn_layout import PlayNetwork
+from output_representation import PlayNetworkPolicyConverter
+
+class MctsTrain: # algorithm for training the model.
+
+    def __init__(self,mcts_simulations):
+
+        # Set the starting fen_string
+        self.start_fen = ("rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP"
+        "/RNBQKBNR w KQkq - 0 1")
+
+        # Set the number of Monte Carlo Simulations
+        self.mcts_simulations = mcts_simulations
+        self.training_examples = []
+
+    def training_episode(self,nnet):
+
+        # At each episode need to set the training example to empty
+        self.training_examples = []
+
+        # The board needs to be initialized ot the starting state
+        board = chess.Board(self.start_fen)
+        fen_string = self.start_fen
+        mcts = Mcts() # assign MCTS object
+        policy_converter = PlayNetworkPolicyConverter()
+
+        while True:
+
+            # Perform mcts simulation
+            for _ in range(self.mcts_simulations):
+                mcts.search(board,nnet)
+
+            # Need to get the moves and policy from the mcts
+            keys,policy = mcts.find_search_probs(fen_string)
```

Comment:
space b/w params

Reply:
Is `policy` the correct name for this? My understanding is you are computing the real search probabilities, while the neural network outputs the policy. Also, `keys` doesn't make it clear what's being returned. Maybe you can change these variable names to `moves, search_probs`, and add a comment that `moves[i]` corresponds to `search_probs[i]`